src/App.tsx

```diff
@@ -1,27 +1,45 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const setGoodsFromServer = (promise:() => Promise<Good[]>) => {
+    promise().then((goodsFromServer) => setGoods(goodsFromServer));
+  };
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => setGoodsFromServer(getAll)}
+      >
+        Load all goods
+      </button>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => setGoodsFromServer(get5First)}
+      >
+        Load 5 first goods
+      </button>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={() => setGoodsFromServer(getRedGoods)}
+      >
+        Load red goods
+      </button>
+
+      <GoodsList goods={goods} />
```

Comment:
In case when you don't have any goods, you don't need to render this component

Reply:
Thanks, fixed!